github-git/2-identify-issue.md

```diff
@@ -0,0 +1,121 @@
+:::
+
+(create-issue)=
+### How to create a new issue
+
+:::{tip}
+Some projects use individual templates for bugs, documentation fixes, and new feature requests. Other projects might have no templates.
+:::
+
+If an issue doesn’t already exist for the thing you'd like to work on, here’s how to create a new one: 
+
+1. Click on the new issues button. If there is a template that you can select for the issue, use one.
+
+1. Write a **clear title** summarizing what you’d like to fix. Some repositories use templates—fill them out if available.
+1. Be **specific about what you'd like to fix**:
+   * Explain the problem or fix you’re proposing.
+   * For bugs, include steps to reproduce the issue.
+   * For documentation, link to the page and describe what you’d like to update or enhance.
+
+
+
+:::{figure} /images/github/open-issue.gif
+:alt: alt text here
+
+:::
+
+:::{admonition} Why Detail Matters
+:class: tip
+
+Maintainers are often volunteers, so the more information you provide, the easier it is for them to support you. Investing time in crafting a clear issue upfront pays off later.
+:::
+
+**Be patient**: Once you have opened your issue, be patient. Maintainers might take time to respond. Their timeline depends on factors like how active the repo is, the size of the maintainer team, and their personal schedules. Responses can take days or weeks.
+
+If you don’t hear back after a few weeks, a polite nudge is fine:
+
+> Hi team 👋, just following up on my issue. Let me know if you’d like me to move forward with a pull request or if there’s anything I should address first.
+
+
+## Tips for submitting good issues
+
+* Search First: Before creating a new issue, check if someone else has already reported it.
+* Be Clear: Provide a detailed, reproducible example when reporting bugs or proposing fixes. Make your GitHub issue title specific enough that people can tell what the issue is about without reading all the details. This helps maintainers and contributors quickly identify overlapping issues.
+
+Examples of Good Issue Titles
+
+* Add: update docstring in the function_name_here() function.
+* Update the module_name.py module with a clearer module docstring.
+* Fix: Typo in the documentation page for doc-page-here.
+* Fix: Bug in xxx module that causes x, y, z to happen.
+
+Examples of less useful issue titles:
+* Update documentation.
+* Fix formatting.
+* Be Constructive: Avoid blaming or harsh criticism. Instead, frame concerns as suggestions.
+    
+Less Constructive: “This feature is broken and useless.”
+
+Constructive: “I encountered an issue when using this feature. Here’s the error and steps to reproduce it.”
+
+## What happens next
+
+If the maintainers invite you to submit a pull request, it's time to:
+
+* [Fork the repository](fork-repo) if you haven't already.
+* [Make your changes on a new branch in your fork.](4-edit-commit-files)
+* [Submit a pull request](5-pull-request) with your updates. 
```

Comment:
This might be something you plan to cover in pull-request, but the Issue and PR should be linked together somehow, whether that is in the PR title or a PR comment mentioning the issue, or an issue comment mentioning the PR.
And once both are open, this guide should take care to be clear about which thread recommended communications should go in.

Reply:
Good idea. I think it would be great to reinforce this a few times. so I'll incorporate it up a bit higher just to raise awareness early on!!

Reply:
ok added - i'll remember to mention this in the pr lesson too!!